pkg/plugin/sdk/v1alpha2/plugin/interface.go

```diff
@@ -14,6 +14,7 @@ import (
 // Plugin determines how an aspect Plugin should be implemented.
 type Plugin interface {
 	BEPEventCallback(event *buildeventstream.BuildEvent) error
+	SetupHook(properties []byte) error
```

Comment:
I don't think it's hooking onto anything. IMO, `Setup` represents the intention better.